phoenix-core/src/main/java/org/apache/phoenix/hbase/index/IndexRegionObserver.java

```diff
@@ -555,33 +570,30 @@ private void prepareIndexMutations(ObserverContext<RegionCoprocessorEnvironment>
                   // add the VERIFIED cell, which is the empty cell
                   Mutation m = next.getFirst().getFirst();
                   boolean rebuild = PhoenixIndexMetaData.isIndexRebuild(m.getAttributesMap());
-                  long ts = getMaxTimestamp(m);
                   if (rebuild) {
                       if (m instanceof Put) {
+                          long ts = getMaxTimestamp(m);
+                          // Remove the empty column prepared by Index codec as we need to change its value
+                          removeEmptyColumn(m, emptyCF, emptyCQ);
```

Comment:
It seems that we first remove the existing "empty" column from an array -- the cell list of the column family then add a new "empty" column. Is it necessary have an optimization here - replace the old cell with the new cell in place in the array?

Reply:
No need to transfer an unnecessary cell mutation over the network. Also, giving the same cell with two different values is confusing, although HBase takes the latest one.

Reply:
That isn't what I meant. I meant overriding the old cell with the new cell in the cell list in place and avoiding removing an element in an array.

Reply:
There is no HBase API to do that as far as I know, that is why I have to remove it.